### PR TITLE
adding within' function

### PR DESCRIPTION
**Pull request by TylerMclaughlin**
Issue #313. 

Incorporating code from @bgold-cosmos following our discussion.   Added comment in code for explanation of what this function does.

### CODE REVIEW

File: Sound/Tidal/Pattern.hs [hunk 1]
```diff
@@ -1116,6 +1116,44 @@ within (s,e) f p = stack [playWhen (\t -> cyclePos t >= s && cyclePos t < e) $ f
                           playWhen (\t -> not $ cyclePos t >= s && cyclePos t < e) $ p
                          ]
 
+{- |
+For many cases, @within'@ will function exactly as within.  
+The difference between the two occurs when applying functions like 'fast' or '<~'. 
+within applies the function to the whole cycle first then keeps notes only in the specified interval. 
+within', however, only sees the notes in the specified interval to which it then applies the function.
```
Comment (telephon) [inline Sound/Tidal/Pattern.hs]: I'd write:

within' keeps notes only in the specified interval first, and then applies the function.

This makes the symmetry clear.

**Comment by TylerMclaughlin** [comment]
@yaxu @telephon what do y'all think?

**Comment by yaxu** [comment]
I'm traveling back from holidays but am in favour! Longer term it would be
good to generalise this sort of thing though, when we understand it better.

**Comment by yaxu** [comment]
Merged - thanks!

**Pull request merged**